### PR TITLE
feat: implements ttl and persist

### PR DESCRIPTION
**Pull request by ksankeerth**
* implement ttl and persist
* docs for ttl, persist, put if exists, and put if not exists

Implements APIs for #517

### CODE REVIEW

**Review by bigboss2063 — CHANGES_REQUESTED** [review]

**Comment by ksankeerth** [comment]
I've made the changes. When you have time, Please check. thanks!

**Comment by bigboss2063** [comment]
Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth

**Comment by bigboss2063** [comment]
> Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth

Like remaining 500ms

**Comment by ksankeerth** [comment]
> > Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth
> 
> Like remaining 500ms

Yeah, Your point is valid. It'll return 0 when casting 500ms to `int`. If You prefer, we may return milliseconds instead of seconds. 

I decided to return in seconds after looking at `TTL` of redis. https://redis.io/commands/ttl/

**Comment by bigboss2063** [comment]
> > > Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth
> > 
> > 
> > Like remaining 500ms
> 
> Yeah, Your point is valid. It'll return 0 when casting 500ms to `int`. If You prefer, we may return milliseconds instead of seconds.
> 
> I decided to return in seconds after looking at `TTL` of redis. https://redis.io/commands/ttl/

I think so. Because our accuracy is only up to seconds, so I think it's okay to return 0 less than 1 second.

**Comment by ksankeerth** [comment]
> > > > Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth
> > > 
> > > 
> > > Like remaining 500ms
> > 
> > 
> > Yeah, Your point is valid. It'll return 0 when casting 500ms to `int`. If You prefer, we may return milliseconds instead of seconds.
> > I decided to return in seconds after looking at `TTL` of redis. https://redis.io/commands/ttl/
> 
> I think so. Because our accuracy is only up to seconds, so I think it's okay to return 0 less than 1 second.

I checked, We'll return 0 for 500ms https://go.dev/play/p/APO-EM5CeUy

File: tx_btree.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ func (tx *Tx) PutIfNotExists(bucket string, key, value []byte, ttl uint32) error
 
 // PutIfExits set the value for a key in the bucket only if the key already exits.
 func (tx *Tx) PutIfExists(bucket string, key, value []byte, ttl uint32) error {
+	return tx.updateIfExists(bucket, key, value, ttl, false)
+}
+
+func (tx *Tx) updateIfExists(bucket string, key, value []byte, ttl uint32, updateTTLOnly bool) error {
```
Comment (bigboss2063) [inline tx_btree.go:68]: We already have helper method named `update` and so on, maybe we can edit and reuse them?

**Comment by bigboss2063** [comment]
> > > > > Hi, just a problem, if the remain TTL is less than 1s, what will be returned? @ksankeerth
> > > > 
> > > > 
> > > > Like remaining 500ms
> > > 
> > > 
> > > Yeah, Your point is valid. It'll return 0 when casting 500ms to `int`. If You prefer, we may return milliseconds instead of seconds.
> > > I decided to return in seconds after looking at `TTL` of redis. https://redis.io/commands/ttl/
> > 
> > 
> > I think so. Because our accuracy is only up to seconds, so I think it's okay to return 0 less than 1 second.
> 
> I checked, We'll return 0 for 500ms https://go.dev/play/p/APO-EM5CeUy

OK, and we should add comment for this.

**Comment by ksankeerth** [comment]
I've updated the PR with your suggestions.

**Review by bigboss2063 — APPROVED** [review]
LGTM!

**Pull request merged**